pandas/tests/extension/test_arrow.py

```diff
@@ -3046,3 +3034,63 @@ def test_factorize_chunked_dictionary():
     exp_uniques = pd.Index(ArrowExtensionArray(pa_array.combine_chunks()))
     tm.assert_numpy_array_equal(res_indices, exp_indicies)
     tm.assert_index_equal(res_uniques, exp_uniques)
+
+
+@pytest.mark.parametrize(
+    "how", ["any", "all", "sum", "prod", "min", "max", "mean", "sem", "std", "var"]
+)
+def test_groupby_reductions(data, how, request):
+    # GH 55234
+    mark_pyarrow = pytest.mark.xfail(
+        raises=pa.ArrowNotImplementedError,
+        reason="no kernel matching input types",
+    )
+    mark_fallback = pytest.mark.xfail(
+        raises=TypeError,
+        reason="agg function failed",
+    )
+    pa_type = data._pa_array.type
+    if pa.types.is_string(pa_type) or pa.types.is_binary(pa_type):
+        if how in ["any", "all", "sem", "std"]:
+            request.node.add_marker(mark_pyarrow)
+        elif how in ["sum", "prod", "mean", "var"]:
+            request.node.add_marker(mark_fallback)
+    elif pa.types.is_duration(pa_type):
+        if how in ["prod", "var"]:
+            request.node.add_marker(mark_fallback)
+    elif pa.types.is_temporal(pa_type):
+        if how in ["any", "all"]:
+            request.node.add_marker(mark_pyarrow)
+        elif how in ["sum", "prod", "var"]:
+            request.node.add_marker(mark_fallback)
```

Comment:
Are there ops in here that don't make sense and so won't ever be fixed? For example, prod of temporal? If that's the case, then I'd prefer to test the failure with `pytest.raises` rather than xfail. I'm thinking we should only be xfailing things which we want to be fixed.

Reply:
good point - updated to `pytest.raises`